### PR TITLE
Add reuse_fd function to allow non-pinned BPF maps to be reused

### PR DESCRIPTION
**Pull request by thesw4rm**
…_fd to allow for non-pinned maps to be used

Signed-off-by: Yadunandan Pillai <ytpillai@pm.me>

### CODE REVIEW

**Comment by thesw4rm** [comment]
Solves #170

File: libbpf-rs/src/map.rs [hunk 1]
```diff
@@ -69,19 +80,14 @@ impl OpenMap {
         if fd < 0 {
             return Err(Error::System(errno::errno()));
         }
-
-        let ret = unsafe { libbpf_sys::bpf_map__reuse_fd(self.ptr, fd) };
+        let reuse_result = self.reuse_fd(fd);
 
         // Always close `fd` regardless of if `bpf_map__reuse_fd` succeeded or failed
```
Comment (insearchoflosttime) [inline libbpf-rs/src/map.rs:85]: nit: can you update this comment as well? It should be "regardless of if `reuse_fd` succeeded or failed"

**Comment by insearchoflosttime** [comment]
Thanks for updating this. Sorry to harp on this, but do you mind consolidating your commit title so that it doesn't trail off into the commit message? One idea would be a title like:

Add reuse_fd function to allow non-pinned BPF maps to be reused

with the message something like:

This patch splits some of the functionality in `reuse_pinned_map` into a separate function `reuse_fd` which can be independently called to reuse an fd for a non-pinned BPF map.

**Comment by thesw4rm** [comment]
> Thanks for updating this. Sorry to harp on this, but do you mind consolidating your commit title so that it doesn't trail off into the commit message?
> 
> One idea would be a title like:
> 
> Add reuse_fd function to allow non-pinned BPF maps to be reused
> 
> with the message something like:
> 
> This patch splits some of the functionality in `reuse_pinned_map` into a separate function `reuse_fd` which can be independently called to reuse an fd for a non-pinned BPF map.

No worries! I respect your dedication to making a quality project. 

I'll implement both changes you mentioned

**Comment by insearchoflosttime** [comment]
Thank you for your contribution to libbpf-rs, @thesw4rm!

**Comment by thesw4rm** [comment]
@insearchoflosttime thank you for the guidance through it!!! 😁

**Comment by DevasiaThomas** [comment]
@insearchoflosttime can we create a release with this change (or with the other changes that have since been done)?

**Comment by insearchoflosttime** [comment]
@DevasiaThomas Yes!

**Comment by insearchoflosttime** [comment]
@DevasiaThomas libbpf-rs and libbpf-cargo crates have been updated to the latest release

**Comment by thesw4rm** [comment]
@insearchoflosttime Thanks!

**Comment by DevasiaThomas** [comment]
> @DevasiaThomas libbpf-rs and libbpf-cargo crates have been updated to the latest release

Thanks :)

**Comment by DevasiaThomas** [comment]
@insearchoflosttime Thank you :)

**Pull request merged**